ddtrace/internal/_encoding.pyx

```
@@ -616,6 +624,44 @@ cdef class MsgpackEncoderV04(MsgpackEncoderBase):
                     return ret
         return 0
 
+    cdef inline int _pack_span_events(self, list span_events):
+        ret = msgpack_pack_array(&self.pk, len(span_events))
+        if ret != 0:
+            return ret
+
+        for event in span_events:
+            # get dict representation of span event
+            d = dict(event)
+            ret = msgpack_pack_map(&self.pk, len(d))
+            if ret != 0:
+                return ret
+
+            for k, v in d.items():
+                # pack the name of a span event
+                ret = pack_text(&self.pk, k)
+                if ret != 0:
+                    return ret
+                if isinstance(v, (int, float)):
+                    ret = pack_number(&self.pk, v)
+                elif isinstance(v, str):
+                    ret = pack_text(&self.pk, v)
+                elif k == "attributes":
+                    # span events can contain attributes, this is analougous to span tags
+                    attributes = v.items()
+                    ret = msgpack_pack_map(&self.pk, len(attributes))
+                    for attr_k, attr_v in attributes:
+                        ret = pack_text(&self.pk, attr_k)
+                        if ret != 0:
+                            return ret
+                        ret = pack_text(&self.pk, attr_v)
+                        if ret != 0:
+                            return ret
+                else:
+                    raise ValueError(f"Failed to encode SpanEvent. {k}={v} contains an unsupported type, {type(v)}")
```

Would this exception be propagated all the way up? If that's the case, how likely is that this shows up in customer's UI?

---

@taegyunkim This exception will show in the logging / telemetry but not in the customer's UI

---

@quinna-h and I did a deep dive into this the other day. These exceptions bubble up through the SpanAggregator even.

We need to find an apprpriate place to catch and log these errors rather than let them bubble up, however this is an existing issue in the encoder and not something new from this PR, so ok to defer as a follow-up.